newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,329 @@
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: 32,
+  },
+};
+
+const useStylesForGridList = makeStyles({
+  root: {
+    overflowX: 'scroll',
+    overflowY: 'hidden',
+    flexWrap: 'nowrap',
+    scrollbarWidth: 'none' /* For Firefox */,
+    '-ms-overflow-style': 'none' /* For Internet Explorer and Edge */,
+    '&::-webkit-scrollbar': {
+      height: 0 /* For Chrome, Safari, and Opera */,
+    },
+  },
+});
+
+const useStylesForGridListItem = makeStyles({
+  root: {
+    width: 'unset !important',
+  },
+  tile: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+});
+
+const Carousel = <ThumbnailType: Thumbnail>({
+  title,
+  items,
+  displayItemTitles = true,
+  tabIndexOffset = 0,
+}: Props<ThumbnailType>) => {
+  const theme = React.useContext(GDevelopThemeContext);
+  const [
+    shouldDisplayLeftArrow,
+    setShouldDisplayLeftArrow,
+  ] = React.useState<boolean>(false);
+  const classesForGridList = useStylesForGridList();
+  const classesForGridListItem = useStylesForGridListItem();
+  const scrollView = React.useRef<?HTMLUListElement>(null);
+  const itemsToDisplay =
+    items ||
+    Array(3)
+      .fill({
+        skeleton: true,
+        title: '',
+        description: '',
+        thumbnail: '',
+      })
+      .map((item, index) => ({ ...item, id: `skeleton${index}` }));
+
+  const windowSize = useResponsiveWindowWidth();
+  const imageHeight = referenceSizesByWindowSize.imageHeight[windowSize];
+  const arrowWidth = referenceSizesByWindowSize.arrowWidth[windowSize];
+  const cellWidth = (16 / 9) * imageHeight;
+  const widthUnit = cellWidth + cellSpacing;
+
+  const cellHeight =
+    imageHeight + (displayItemTitles ? titleHeight + spacerSize : 0);
+
+  const renderImage = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): React.Node => (
+      <img
+        src={item.thumbnailUrl}
+        style={{
+          ...styles.image,
+          height: imageHeight,
+          minHeight: imageHeight,
+          width: cellWidth,
+        }}
+        alt={item.title}
+        title={item.title}
+      />
+    ),
+    [cellWidth, imageHeight]
+  );
+
+  const renderThumbnail = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!item.skeleton && !item.link && !item.thumbnailUrl) return null;
+      if (item.link) {
+        return (
+          <a href={item.link} target="_blank">
+            {renderImage(item)}
+          </a>
+        );
+      }
+      if (item.thumbnailUrl) {
+        return renderImage(item);
+      }
+      if (item.skeleton) {
+        return (
+          <Skeleton variant="rect" height={imageHeight} width={cellWidth} />
+        );
+      }
+    },
+    [renderImage, cellWidth, imageHeight]
+  );
+
+  const renderItemTitle = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!displayItemTitles) return null;
+      return (
+        <>
+          <Spacer />
```

Comment:
spacer looks weird here, should it be more where the function `renderItemTitle` is used?

Reply:
Well, if we don't render the title, there is a spacer although it is useless. So I figured I would put it here